### PR TITLE
Add Roger Router

### PR DESCRIPTION
**Pull request by janbrummer**

### CODE REVIEW

**Comment by LeandroStanger** [comment]
bot, build org.tabos.roger

File: org.tabos.roger.json [hunk 1]
```diff
@@ -0,0 +1,259 @@
+{
+    "app-id" : "org.tabos.roger",
+    "runtime" : "org.gnome.Platform",
+    "runtime-version" : "3.32",
```
Comment (LeandroStanger) [inline org.tabos.roger.json]: Update runtime to 3.34

File: org.tabos.roger.json [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+    "runtime-version" : "3.34",
+    "sdk" : "org.gnome.Sdk",
+    "command" : "roger",
+    "finish-args" : [
+        "--socket=x11",
+        "--share=ipc",
+        "--socket=wayland",
+        "--share=network",
+        "--filesystem=host",
+        "--filesystem=xdg-run/dconf",
```
Comment (bilelmoussaoui) [inline org.tabos.roger.json]: dconf permissions are not needed with 3.34/19.08

**Comment by LeandroStanger** [comment]
bot, build org.tabos.roger

**Comment by bilelmoussaoui** [comment]
bot, build org.tabos.roger

File: org.tabos.roger.json [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+    "sdk" : "org.gnome.Sdk",
+    "command" : "roger",
+    "finish-args" : [
+        "--socket=x11",
+        "--share=ipc",
+        "--socket=wayland",
+        "--share=network",
+        "--filesystem=host",
+        "--filesystem=~/.config/dconf:ro",
+        "--talk-name=ca.desrt.dconf",
```
Comment (TingPing) [inline org.tabos.roger.json]: Remove all DConf permissions.

**Comment by janbrummer** [comment]
How should i handle platform specific faults as in aarch64? According to internet it is an issue in libpng and that's part of the runtime...

**Comment by bilelmoussaoui** [comment]
> How should i handle platform specific faults as in aarch64? According to internet it is an issue in libpng and that's part of the runtime...

You could ignore aarch64 for now and report that issue upstream/investigate that further later.

**Comment by bilelmoussaoui** [comment]
bot, build org.tabos.roger

File: org.tabos.roger.json [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+    "runtime-version" : "3.34",
+    "sdk" : "org.gnome.Sdk",
+    "command" : "roger",
+    "finish-args" : [
+        "--socket=x11",
+        "--share=ipc",
+        "--socket=wayland",
+        "--share=network",
+        "--filesystem=host",
+        "--env=DCONF_USER_CONFIG_DIR=.config/dconf",
```
Comment (bilelmoussaoui) [inline org.tabos.roger.json]:
```suggestion
```

**Comment by nedrichards** [comment]
In order to ignore it you can add a flathub.json file that looks/works a bit like this https://github.com/flathub/com.spotify.Client/blob/master/flathub.json

**Comment by janbrummer** [comment]
Thanks, i will try to modify ghostscript flags first. In case it is not working i will go with your proposed solution.

**Comment by bilelmoussaoui** [comment]
bot, build org.tabos.roger

You can run builds yourself too. It doesn't require any permissions.

**Comment by janbrummer** [comment]
bot, build org.tabos.roger

**Comment by janbrummer** [comment]
bot, build org.tabos.roger

File: org.tabos.roger.json [hunk 5]
```diff
@@ -0,0 +1,270 @@
...
+                 }
+            ]
+        },
+        {
+            "name" : "roger",
+            "config-opts" : [
+                "--libdir=/app/lib",
+                "--buildtype=release"
+            ],
+            "only-arches": ["x86_64"],
```
Comment (bilelmoussaoui) [inline org.tabos.roger.json]: You need a flathub.json file instead of using only-arches here.

**Comment by janbrummer** [comment]
bot, build org.tabos.roger

**Comment by nedrichards** [comment]
This crashes and doesn't launch for me out of the box. Is there more I need to do to to run it? If so, we should mention that in https://gitlab.com/tabos/rogerrouter/blob/master/roger/org.tabos.roger.appdata.xml.in

**Comment by janbrummer** [comment]
It shouldn't crash and now it does. As it is something deep down in gtk using gs!? (also png) it must be something with either the new runtime or the 3rd party libs. Compiling it without flatpak works fine. I'll debug it.

**Comment by nedrichards** [comment]
@janbrummer you can also build it against a 3.32 in the short term if it works there.

**Comment by janbrummer** [comment]
bot, build org.tabos.roger

**Comment by janbrummer** [comment]
FTR: Bundled ghostscript has bundled it's own libraries of e.g. libpng. Those libraries have been used during linking and also caused the crash. Furthermore it seems to be responsible for aarch64 build break.

**Comment by janbrummer** [comment]
bot, build org.tabos.roger

**Comment by janbrummer** [comment]
Ok, what's wrong here? I don't see it. submodule is there and it works fine locally.

**Comment by janbrummer** [comment]
bot, build org.tabos.roger

**Comment by bilelmoussaoui** [comment]
It seems we have an issue here see https://github.com/flathub/flathub/issues/1170

**Comment by bilelmoussaoui** [comment]
bot, build org.tabos.roger

**Comment by bilelmoussaoui** [comment]
bot, build org.tabos.roger

**Comment by janbrummer** [comment]
bot, build org.tabos.roger

**Comment by janbrummer** [comment]
bot, build org.tabos.roger

**Comment by janbrummer** [comment]
bot, build org.tabos.roger

**Comment by janbrummer** [comment]
Ok, i will not try to fix aarch64 for now -.-

**Comment by janbrummer** [comment]
bot, build org.tabos.roger

**Comment by bilelmoussaoui** [comment]
Repository has been created: https://github.com/flathub/org.tabos.roger


Thanks!

**Comment by bilelmoussaoui** [comment]
Let me know if anyone else needs write access to the repository.

**Pull request closed without merging**